### PR TITLE
Improve .bib from .pdf

### PR DESCRIPTION
**Pull request by koppor**
Follow-up to https://github.com/JabRef/jabref/pull/11156

Now covers more cases and new menu entry:

**Update**

grafik

### Mandatory checks

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/gui/frame/MainMenu.java [hunk 1]
```diff
@@ -303,6 +314,12 @@ private void createMenu() {
 
                 factory.createMenuItem(StandardActions.REDOWNLOAD_MISSING_FILES, new RedownloadMissingFilesAction(stateManager, dialogService, preferencesService.getFilePreferences(), taskExecutor))
         );
+
+        EasyBind.subscribe(preferencesService.getGrobidPreferences().grobidEnabledProperty(), enabled -> {
+            newLibraryFromPdfMenuItemOnline.setVisible(enabled);
+            newLibraryFromPdfMenuItemOffline.setVisible(!enabled);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/frame/MainMenu.java:320]: offline should be also available even if I have grobid enabled

Comment (koppor) [reply src/main/java/org/jabref/gui/frame/MainMenu.java:320]: Done 😅

File: src/main/java/org/jabref/gui/maintable/NewLibraryFromPdfAction.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        FileDialogConfiguration fileDialogConfiguration = builder.build();
+
+        LOGGER.trace("Opening file dialog with configuration: {}", fileDialogConfiguration);
+
+        dialogService.showFileOpenDialog(fileDialogConfiguration).ifPresent(path -> {
+            LOGGER.trace("Selected file: {}", path);
+            Callable<ParserResult> parserResultCallable = getParserResultCallable(path);
+            BackgroundTask.wrap(parserResultCallable)
+                          .withInitialMessage(Localization.lang("Processing PDF(s)"))
+                          .onFailure(failure -> Platform.runLater(() -> dialogService.showErrorDialogAndWait(failure)))
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/maintable/NewLibraryFromPdfAction.java:79]: onFailure  and onSucccces are alwas execute on the fx thraed, no wrapping necessary

**Comment by Siedlerchr** [comment]
Has issues with parsing authors for inProceeedings, I tested with. https://www.usenix.org/system/files/conference/osdi14/osdi14-paper-yuan.pdf

grafik

**Review by Siedlerchr — APPROVED** [review]

**Review by Siedlerchr — APPROVED** [review]

**Pull request merged**